common/src/main/java/org/keycloak/common/util/CollectionUtil.java

```diff
@@ -79,14 +66,4 @@ public static boolean isNotEmpty(Collection<?> collection) {
         return !isEmpty(collection);
     }
 
-    public static <T> Set<T> intersection(Collection<T> col1, Collection<T> col2) {
-        if (isEmpty(col1) || isEmpty(col2)) return Collections.emptySet();
-
-        final Collection<T> iteratorCollection = col1.size() <= col2.size() ? col1 : col2;
-        final Collection<T> searchCollection = iteratorCollection.equals(col1) ? col2 : col1;
-
-        return iteratorCollection.stream()
-                .filter(searchCollection::contains)
-                .collect(Collectors.toSet());
-    }
```

Comment:
This is the only breaking change (if I haven't overlooked something). I would probably go ahead and just remove it. But I'd suggest adding a note to the upgrading guide (with a suggested code to use instead).